variables.tf

```diff
@@ -8,6 +8,46 @@ variable "create_asg" {
   default     = true
 }
 
+variable "create_asg_with_initial_lifecycle_hook" {
+  description = "Create an ASG with intial"
+  default     = false
+}
+
+variable "initial_lifecycle_hook_name" {
+  description = "Initial lifecycle hook name required parameter"
+  default     = ""
+}
+
+variable "initial_lifecycle_hook_transition" {
+  description = "Initial Lifecycle hook transtion type"
+  default     = ""
+}
+
+variable "initial_lifecycle_hook_default_result" {
+  description = "Defines the default action the autoscaler takes when the lifecyle hook timesout"
```

Comment:
"Defines the action the Auto Scaling group should take when the lifecycle hook timeout elapses or if an unexpected failure occurs. The value for this parameter can be either CONTINUE or ABANDON. The default value for this parameter is ABANDON."